### ISSUE TITLE
Feature: adaptive height

### ISSUE THREAD

**Issue by ajoshi0** [FormidableLabs/nuka-carousel#285]
## Updated ticket

- [x] ~~Get adaptive height in `3.x` branch~~ (fixed in #291)
- [x] Get adaptive height port of #291 into `master` branch after #293 lands.

## Arunesh's Original Description

please check these pr's and merge them which one is appropriate

https://github.com/FormidableLabs/nuka-carousel/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+height

Currently slick supports adaptive height but NUKA does not.

cc @kenwheeler @ryan-roemer

**Comment by johnking** [comment]
I just started using the tool, and spent a few hours on the the height issue.  Good to know it is on the milestone.

**Comment by Nehle** [comment]
@ryan-roemer -- trying to build something off of the current PRs that are out there, but it seems the last PR merged removed the testing and broke the demo. Any ETA on when those will be working again so I can proceed?

cc @ajoshi0 @kenwheeler @alexlande

**Comment by ryan-roemer** [comment]
@Nehle -- @carlospaelinck is actively working on getting the entire infrastructure in modern shape and I think https://github.com/FormidableLabs/nuka-carousel/pull/289 gets us back in "open for business" shape.

Assuming that's the case (is that right Carlos?), we should be ready to pick one of these approaches and have someone open a new PR to fix + add regression tests (using Jest now)

**Comment by Nehle** [comment]
Excellent, I'll rebase my local branch on that. Thanks

**Comment by carlospaelinck** [comment]
Yeah. This PR will finish the tooling. We will also be migrating the code base away from a Mixin-based approach to a modern style as well. So things will be in flux for a bit.

**Comment by ajoshi0** [comment]
@carlospaelinck can we try to keep prop names and functionality as close as possible to current one if possible

**Comment by carlospaelinck** [comment]
Yes, the goal is an internal refactor.

**Comment by Nehle** [comment]
Is there any chance we can get a `3.x` branch to push the height feature to while we're waiting for the upgrade to be completed?

**Comment by ryan-roemer** [comment]
@kenwheeler @carlospaelinck -- How far off is https://github.com/FormidableLabs/nuka-carousel/pull/289 from merging? I _think_ once we land that `master` is back open for business right?

**Comment by carlospaelinck** [comment]
@Nehle We cut a 3.x branch if you want to work off there. For master we are doing some heavy refactoring moving away from `createReactClass` and moving from `react-tween-state` to `react-move`.

**Comment by Nehle** [comment]
Okay, let me prepare a PR for the 3.x branch and I can take responsibility for porting it over to the new version once that's ready

**Comment by ryan-roemer** [comment]
Alrighty!

Carlos approved and I merged and released #291 to `3.x` branch

@Nehle -- I've refactored this ticket to now be "port that to `master`" branch, so that your change will actually be kept in "real" nuka we're releasing probably next week. Appreciate a PR porting over this work to new `master` after #293 lands from you, @tchon or @ajoshi0 

Thanks!

**Comment by ryan-roemer** [comment]
*Update* -- Carlos and I have finished all the `v4.0` milestone work on our end (React upgrades, infrastructure updates too -- e.g., we're publishing `es/` stuff now that modern webpack consumes).

The *last* thing we're waiting on for 4.0 release is the port of @Nehle 's fine work to `master` branch.

@Nehle @ajoshi0 @tchon -- Can you folks help us out with a port PR if you've got a moment? Thanks!

**Comment by ryan-roemer** [comment]
**Update** -- Decorator deprecation coming in https://github.com/FormidableLabs/nuka-carousel/pull/299 so maybe wait until after that (shouldn't impact the PR really though...)

**Comment by ryan-roemer** [comment]
Alrighty, #299 has landed in `master` and the last substantive thing we need for https://github.com/FormidableLabs/nuka-carousel/milestones/4.0 is this work.

@Nehle @ajoshi0 @tchon -- Can y'all confirm availability to help and likely timeline for the port to land? Thanks!

**Comment by carlospaelinck** [comment]
I can be responsible for porting this over to 4.0
